tests/pds/pds_storagesize_increase_test.go

```diff
@@ -0,0 +1,901 @@
+								CpuLimit:       *resConfigModel.CpuLimit,
+								CpuRequest:     *resConfigModel.CpuRequest,
+								DataServiceID:  dataserviceID,
+								MemoryLimit:    *resConfigModel.MemoryLimit,
+								MemoryRequest:  *resConfigModel.MemoryRequest,
+								Name:           newTemplateName,
+								StorageRequest: params.StorageConfigurations.NewStorageSize,
+								FsType:         *stConfigModel.Fs,
+								ReplFactor:     *stConfigModel.Repl,
+								Provisioner:    *stConfigModel.Provisioner,
+								Secure:         false,
+								VolGroups:      false,
+							}
+							stConfigModelUpdated, resConfigModelUpdated, err = controlPlane.CreateCustomResourceTemplate(tenantID, updatedTemplateConfig)
+							log.FailOnError(err, "Unable to update template")
+							log.InfoD("Successfully updated the template with ID- %v", resConfigModelUpdated.GetId())
+							newResourceTemplateID = resConfigModelUpdated.GetId()
+							newStorageTemplateID = stConfigModelUpdated.GetId()
+							stIDs = append(stIDs, newStorageTemplateID)
+							resIds = append(resIds, newResourceTemplateID)
+						})
+						stepLog = "Apply updated template to the dataservice deployment"
+						Step(stepLog, func() {
+							log.InfoD(stepLog)
+							if appConfigID == "" {
+								appConfigID, err = controlPlane.GetAppConfTemplate(tenantID, ds.Name)
+								log.FailOnError(err, "Error while fetching AppConfigID")
+							}
+							updatedDeployment, err = dsTest.UpdateDataServices(deployment.GetId(),
+								appConfigID, deployment.GetImageId(),
+								int32(ds.Replicas), newResourceTemplateID, params.InfraToTest.Namespace)
+							log.FailOnError(err, "Error while updating dataservices")
+							Step("Validate Deployments after template update", func() {
+								err = dsTest.ValidateDataServiceDeployment(updatedDeployment, namespace)
+								log.FailOnError(err, "Error while validating dataservices")
+								log.InfoD("Data-service: %v is up and healthy", ds.Name)
+								updatedDepList = append(updatedDepList, updatedDeployment)
+								updatedPvcSize, err = GetVolumeCapacityInGB(namespace, updatedDeployment)
+								log.InfoD("Updated Storage Size is- %v", updatedPvcSize)
+								dsEntity = restoreBkp.DSEntity{
+									Deployment: updatedDeployment,
+								}
+							})
+							stepLog = "Validate Workload is running after storage resize"
+							Step(stepLog, func() {
+								err = k8sApps.ValidateDeployment(workloadDep, timeOut, 10*time.Second)
+								log.FailOnError(err, "Workload is not running after Storage Size Increase")
+							})
+							stepLog = "Verify storage size before and after storage resize - Verify at STS, PV,PVC level"
+							Step(stepLog, func() {
+								_, _, config, err := pdslib.ValidateDataServiceVolumes(updatedDeployment, ds.Name, newResourceTemplateID, newStorageTemplateID, params.InfraToTest.Namespace)
+								log.FailOnError(err, "error on ValidateDataServiceVolumes method")
```

Comment:
can we have a common function for SSI validation (before and after) where we have all the `dash.VerifyFatal` happens there ?
refer `ValidateDeployments` method